### PR TITLE
Add option to supply sunz-threshold applied in Pyspectral

### PR DESCRIPTION
**Pull request by adybbroe**
Add option to supply sunz-threshold applied in Pyspectral for the emissive/reflectance separation of the 3.x band

Signed-off-by: Adam.Dybbroe <a000680@c21856.ad.smhi.se>

 - [ ] Closes #xxxx 
 - [x] Tests added 
 - [x] Tests passed 
 - [ ] Passes ``flake8 satpy`` 
 - [ ] Fully documented 
 - [ ] Add your name to `AUTHORS.md` if not there already

### CODE REVIEW

File: satpy/composites/__init__.py [hunk 1]
```diff
@@ -647,6 +659,17 @@ def _get_reflectance(self, projectables, optional_datasets):
 class NIREmissivePartFromReflectance(NIRReflectance):
     """Get the emissive par of NIR bands."""
 
+    def __init__(self, **kwargs):
```
Comment (mraspaud) [inline satpy/composites/__init__.py:662]: You can make max_sza explicit:
```suggestion
    def __init__(self, max_sza=None, **kwargs):
```

File: satpy/composites/__init__.py [hunk 2]
```diff
@@ -589,6 +589,17 @@ def __call__(self, projectables, optional_datasets=None, **info):
 class NIRReflectance(CompositeBase):
     """Get the reflective part of NIR bands."""
 
+    def __init__(self, **kwargs):
```
Comment (mraspaud) [inline satpy/composites/__init__.py:592]: You can make max_sza explicit:
```suggestion
    def __init__(self, max_sza=None, **kwargs):
```

File: satpy/composites/__init__.py [hunk 3]
```diff
@@ -589,6 +589,17 @@ def __call__(self, projectables, optional_datasets=None, **info):
...
+        """Collect custom configuration values.
+
+        Args:
+            max_sza (float): Maximum solar zenith angle in degrees that is
+                considered valid and correctable. Default 95.0.
+
+        """
+        self.sunz_threshold = kwargs.get('sunz_threshold')
```
Comment (mraspaud) [inline satpy/composites/__init__.py:600]:
```suggestion
        self.sunz_threshold = max_sza
```

File: satpy/composites/__init__.py [hunk 4]
```diff
@@ -647,6 +659,17 @@ def _get_reflectance(self, projectables, optional_datasets):
...
```
Comment (mraspaud) [inline satpy/composites/__init__.py:670]:
```suggestion
        self.sunz_threshold = max_sza
```

File: satpy/composites/__init__.py (hunk 3 above)
Comment (mraspaud) [reply satpy/composites/__init__.py:600]: oh and there seems to be a missmatch between the docstring and actual keyword name

File: satpy/composites/__init__.py (hunk 4 above)
Comment (adybbroe) [reply satpy/composites/__init__.py]: I didn't see this parameter `max_sza` actually. As it said default=95 I thought it must be something else. I have no idea what that might have been used for, but think it came from a cut and paste (error) from the sunz-corrector. I deleted it and kept sunz_threshold.

File: satpy/composites/__init__.py (hunk 3 above)
Comment (adybbroe) [reply satpy/composites/__init__.py]: See below and latest push

File: satpy/etc/composites/visir.yaml [hunk 5]
```diff
@@ -32,7 +33,8 @@ modifiers:
     optional_prerequisites:
     - solar_zenith_angle
     - 13.4
-
+    sunz_threshold: 85.0
+
```
Comment (mraspaud) [inline satpy/etc/composites/visir.yaml:37]:
```suggestion

```

**Pull request merged**